server/auth/auth.go

```diff
@@ -694,7 +694,35 @@ func (auth *AuthManager) Notify(acctID account.AccountID, msg *msgjson.Message,
 
 // Penalize signals that a user has broken a rule of community conduct, and that
 // their account should be penalized.
-func (auth *AuthManager) Penalize(user account.AccountID, rule account.Rule) error {
+func (auth *AuthManager) Penalize(user account.AccountID, rule account.Rule, extraDetails string) error {
+	// Notify user of penalty.
+	details := "Ordering has been suspended for this account. Contact the exchange operator to reinstate privileges."
+	if auth.anarchy {
+		details = "You were penalized but the penalty will not be counted against you."
+	}
+	details = fmt.Sprintf("%s\nRule Details: %s\n%s", details, rule.Description(), extraDetails)
```

Comment:
Because `Penalize` does `client := auth.user(user)`, we have at least one place that manually just does `client.suspended = true; auth.storage.CloseAccount()` manually to avoid it.  As a result, `recordOrderDone`, where the cancellation rate penalty may take affect, is going to lack this notification.
I'd be OK resolving that by changing `recordOrderDone` to not do `client.mtx.Lock/Unlock`, and instead use a new mutex `clientInfo.recentOrdersMtx` in it's place so it can at least do the `add` and `checkCancelRate` operations atomically, then use `Penalize` directly instead of the manual operations it does now.

Reply:
Does this work? The two are still running atomically. There is a small gap before penalizing...